test/com/amazon/ion/CloneTest.java

```diff
@@ -160,4 +160,67 @@ public void testDifferentValueFactoryCloneWithUnknownFieldNameText()
         // If we don't fail we should at least retain the SID.
         assertEquals(99, copy.iterator().next().getFieldNameSymbol().getSid());
     }
+
+    @Test
+    public void cloneEmptyContainer()
+    {
+        IonStruct original = system().newEmptyStruct();
+        IonStruct clone = original.clone();
+        assertTrue(clone.isEmpty());
+        assertEquals(original, clone);
+    }
```

Comment:
We need in this test suite non-reference-equality assertions and some clarification of behavior around modification.

Reply:
Please expand on what you mean by this, maybe with an example.

Reply:
So many of these tests are satisfiable by a trivial clone implementation, `return this;`. 
When asserting anything about clones, we should also assert that they are not the same object. Here's a take on CloneTest that adds that assertion, cleans up some other assertions, and detangles from IonTestCase/Junit4 Assert: https://gist.github.com/jobarr-amzn/77c9141ee9106d7c956d8cb0f9e13db8

For clarifying behavior around mutation, we can assert that modifying the original does not modify the clone and vice versa. I didn't do that because I don't know how in the general case and doing it non-generally would be more work :)

I'm fine to defer that with a comment or issue.

Reply:
Looks good to me, added. For verifying mutation, for now I've added a specific test for this and a comment about generalizing it.